compiler/src/iree/compiler/Codegen/Common/AddFastMathFlags.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/PassDetail.h"
+#include "iree/compiler/Codegen/Common/Passes.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+
+#define DEBUG_TYPE "iree-codegen-add-fast-math-flags"
+
+using namespace mlir;
+using namespace mlir::iree_compiler;
+
+/// Add `contract` FMF to operations that support it.
+static void addContractFMF(Operation *op) {
+  LLVM::FastmathFlags contract = LLVM::FastmathFlags::contract;
+  TypeSwitch<Operation *>(op)
+      .Case<LLVM::FMulOp>(
+          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); })
+      .Case<LLVM::FAddOp>(
+          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); })
+      .Case<LLVM::FSubOp>(
+          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); })
+      .Case<LLVM::FNegOp>(
+          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); });
```

Comment:
I'm not fully sure, but I think we can merge these cases into a single `Case`. Then we can save 6 lines.

```suggestion
      .Case<LLVM::FMulOp, LLVM::FAddOp, LLVM::FSubOp, LLVM::FNegOp>(
          [&](auto llvmOp) { llvmOp.setFastmathFlags(contract); })
```